terraform/environments/dev/remote_state/main.tf

```diff
@@ -0,0 +1,55 @@
+terraform {
+  required_providers {
+    aws = {
+      source = "hashicorp/aws"
+    }
+  }
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+# "State store for terraform with S3"
+resource "aws_s3_bucket" "sdc_state_terraform" {
+  bucket = "sdc-dev-terraform-state"
+  # acl?
+  versioning {
+    enabled = true
+  }
+
+  lifecycle {
+    prevent_destroy = true
+  }
+
+  tags = {
+    name = "Terraform Remote S3 State Store"
+  }
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+}
+
+# State locks for terraform with dynamo
+resource "aws_dynamodb_table" "sdc_terraform_state_lock" {
```

Comment:
I don't think dynamodb has a forever free tier, but I guess it is pretty cheap 👀

Reply:
ah didn't know about that. I'm not sure there is a workaround? This is what everybody does for terraform :(

Reply:
meh it's probably fine. I think the usage should just be 1-3 dollars/month at worst hopefullly. Do keep an eye on the billing though 😉